pkg/entrypoint/entrypointer.go

```diff
@@ -280,27 +283,39 @@ func (e Entrypointer) Go() error {
 	}
 
 	if e.ResultExtractionMethod == config.ResultExtractionMethodTerminationMessage {
+		// step artifacts
 		fp := filepath.Join(e.StepMetadataDir, "artifacts", "provenance.json")
+		artifacts, err := readArtifacts(fp, result.StepArtifactsResultType)
+		if err != nil {
+			logger.Fatalf("Error while handling step artifacts: %s", err)
+		}
+		output = append(output, artifacts...)
 
-		artifacts, err := readArtifacts(fp)
+		artifactsDir := pipeline.ArtifactsDir
+		// task artifacts
+		if e.ArtifactsDirectory != "" {
+			artifactsDir = e.ArtifactsDirectory
+		}
+		fp = filepath.Join(artifactsDir, "provenance.json")
```

Comment:
We might want to make this an exported constant at least. Also, naive question, why the name `provenance.json` ?

Reply:
agreed about exported const.
provenance.json because it contains information about the location of the artifacts (i.e. Uri and digest). We can certainly change the name to something else. We thought that it was accurate.

Reply:
To make fp a constant value, it would require artifactsDir to be a constant first, but it needs to be a variable as we need to overwrite this value for test purpose.  This is the same pattern used for resultDir https://github.com/tektoncd/pipeline/blob/34c3828589a0e37ef2d4adaa2f170e4ea1fc797c/pkg/entrypoint/entrypointer.go#L267-L273